conjur/login/login_controller.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+"""
+LoginController module
+
+This module is the controller that facilitates all login actions
+required to successfully configure the netrc
+"""
+
+# Builtins
+import getpass
+import logging
+import sys
+
+from conjur.init import ConjurrcData
+from conjur.utils import Utils
+import conjur.constants
+
+class LoginController:
+    """
+    LoginController
+
+    This class represents the Presentation Layer for the LOGIN command
+    """
+    password = None
+    credential_data = None
+    login_logic = None
+    ssl_verify = None
+
+    def __init__(self, ssl_verify, password, credential_data, login_logic):
+        self.ssl_verify = ssl_verify
+        if self.ssl_verify is False:
+            Utils.insecure_message()
+
+        self.password = password
+        self.credential_data = credential_data
+        self.login_logic = login_logic
+
+    def load(self):
+        """
+        Method that facilitates all method calls in this class
+        """
+        self.get_username()
+        self.password = self.get_password()
+        loaded_conjurrc = self.load_conjurrc_data()
+        self.get_api_key(self.ssl_verify, loaded_conjurrc)
+        # pylint: disable=logging-fstring-interpolation
+        logging.debug(f"Saving credentials to '{conjur.constants.DEFAULT_NETRC_FILE}'")
+        self.login_logic.save(self.credential_data)
+        # pylint: disable=logging-fstring-interpolation
+        logging.debug("Credentials successfully written to " \
+                      f"'{conjur.constants.DEFAULT_NETRC_FILE}'")
+        sys.stdout.write("Successfully logged in to Conjur!\n")
+
+    def get_username(self):
+        """
+        Method to fetch the username if the user did not provide one
+        """
+        if self.credential_data.login is None:
+            # pylint: disable=logging-fstring-interpolation
+            self.credential_data.login = input("Enter your login name to log into Conjur: ").strip()
+            if self.credential_data.login == '':
+                # pylint: disable=raise-missing-from
+                raise RuntimeError("Error: Login name is required")
+
+    def get_password(self):
+        """
+        Method to fetch the password from the user attempting to login
+        """
+        if self.password is None:
+            # pylint: disable=line-too-long
+            return getpass.getpass(prompt=f"Please enter {self.credential_data.login}'s password (it will not be echoed): ")
+        return self.password
```

Comment:
no because even if it is None, we will still attempt to make a login request. We could block empty input but didn't think it was necessary since it is only a single login request that is made